File: app/src/main/java/com/system/m4/views/home/MainPresenter.java
package com.system.m4.views.home;

import com.system.m4.infrastructure.JavaUtils;
import com.system.m4.views.vos.TagVO;

import java.util.Calendar;

/**
 * Created by eferraz on 29/07/17.
 */

public class MainPresenter implements MainContract.Presenter {

    private MainContract.View mView;

    public MainPresenter(MainContract.View mView) {
        this.mView = mView;
        configureTitle(1);
    }

    @Override
    public void requestTransactionManager() {
        mView.requestTransactionManagerDialog(); // Colocar PLaceholder e remover if list == null do adapter
    }

    @Override
    public void requestTransactionDialog(TagVO vo) {
        mView.showTransactionDialog(vo);
    }

    @Override
    public void configureTitle(int position) {
        Calendar calendar = Calendar.getInstance();
        calendar.set(Calendar.MONTH, position - MainPageAdapter.PAGE_MIDDLE);
        mView.setMainTitle(JavaUtils.DateUtil.format(calendar.getTime(), JavaUtils.DateUtil.MMMM_DE_YYYY));
    }

}

File: app/src/main/java/com/system/m4/kotlin/presenters/MainPresenter.kt
package com.system.m4.kotlin.presenters

import com.system.m4.infrastructure.JavaUtils
import com.system.m4.kotlin.contracts.MainContract
import com.system.m4.kotlin.view.adapters.MainPageAdapter
import com.system.m4.labs.vos.TagVO
import java.util.*

/**
 * Created by eferraz on 29/07/17.
 */

class MainPresenter(private val mView: MainContract.View) : MainContract.Presenter {

    init {
        configureTitle(MainPageAdapter.PAGE_MIDDLE)
    }

    override fun requestTransactionManager() {
        mView.requestTransactionManagerDialog() // Colocar PLaceholder e remover if list == null do adapter
    }

    override fun requestTransactionDialog(vo: TagVO) {
        mView.showTransactionDialog(vo)
    }

    override fun configureTitle(position: Int) {
        val calendar = Calendar.getInstance()
        calendar.add(Calendar.MONTH, position - MainPageAdapter.PAGE_MIDDLE)
        mView.setMainTitle(JavaUtils.DateUtil.format(calendar.time, JavaUtils.DateUtil.MMMM_DE_YYYY))
    }

}
